libs/app-mtp/src/lib/dotnet-mtp/dotnet-mtp.ts

```diff
@@ -118,10 +120,27 @@ export class DotnetMtp implements MtpInterface {
       : Result.success({ progress: transactionStatus.progress })
   }
 
+  async cancelUpload(
+    data: UploadTransactionData
+  ): Promise<ResultObject<CancelUploadResultData>> {
+    if (this.uploadFileTransactionStatus[data.transactionId] !== undefined) {
+      this.abortController?.abort()
+      console.log(
+        `${PREFIX_LOG} Canceling upload for transactionId ${data.transactionId}, signal abort status: ${this.abortController?.signal.aborted}`
+      )
+      return Result.success({})
+    } else {
+      return Result.failed({
+        type: MTPError.MTP_TRANSACTION_NOT_FOUND,
+      } as AppError)
```

Comment:
I wonder if in this case we should return an error-like response when transaction wasn't found. After all, the transaction is gone, and that's exactly what user was expecting.